show/main.py

```diff
@@ -2082,6 +2082,134 @@ def ztp(status, verbose):
     run_command(cmd, display_cmd=verbose)
 
 
+#
+# 'bmp' group ("show bmp ...")
+#
+@cli.group(cls=clicommon.AliasedGroup)
+def bmp():
+    """Show details of the bmp dataset"""
+    pass
+
+
+# 'bgp-neighbor-table' subcommand ("show bmp bgp-neighbor-table")
+@bmp.command('bgp-neighbor-table')
+@clicommon.pass_db
+def bmp_neighbor_table(db):
+    """Show bmp bgp-neighbor-table information"""
+    bmp_headers = ["Neighbor_Address", "Peer_Address", "Peer_ASN", "Peer_RD", "Peer_Port",
+                   "Local_Address", "Local_ASN", "Local_Port", "Advertised_Capabilities", "Received_Capabilities"]
+
+    # BGP_NEIGHBOR_TABLE|10.0.1.2
+    bmp_keys = db.db.keys(db.db.STATE_DB, "BGP_NEIGHBOR_TABLE|*")
+
+    click.echo("Total number of bmp neighbors: {}".format(0 if bmp_keys is None else len(bmp_keys)))
+
+    bmp_body = []
+    if bmp_keys is not None:
+        for key in bmp_keys:
+            values = db.db.get_all(db.db.STATE_DB, key)
+            bmp_body.append([
+                values["peer_addr"],  # Neighbor_Address
+                values["peer_addr"],
+                values["peer_asn"],
+                values["peer_rd"],
+                values["peer_port"],
+                values["local_addr"],
+                values["local_asn"],
+                values["local_port"],
+                values["sent_cap"],
+                values["recv_cap"]
+            ])
+
+    click.echo(tabulate(bmp_body, bmp_headers))
+
+
+# 'bmp-rib-out-table' subcommand ("show bmp bgp-rib-out-table")
+@bmp.command('bgp-rib-out-table')
+@clicommon.pass_db
+def bmp_rib_out_table(db):
+    """Show bmp bgp-rib-out-table information"""
+    bmp_headers = ["Neighbor_Address", "NLRI", "Origin", "AS_Path", "Origin_AS", "Next_Hop", "Local_Pref",
+                   "Originator_ID",  "Community_List", "Ext_Community_List"]
+
+    # BGP_RIB_OUT_TABLE|192.181.168.0/25|BGP_NEIGHBOR|10.0.0.59
```

Comment:
>BGP_NEIGHBOR [](http://example.com/codeflow?start=41&length=12)

The schema design is weird, BGP_NEIGHBOR is hardcoded string, and can be removed.

Reply:
changed into BGP_RIB_OUT_TABLE|192.181.168.0/25|10.0.0.59, also changed https://github.com/sonic-net/sonic-bmp/pull/7 for db population.